hub/core/query/filter.py

```diff
@@ -165,23 +166,26 @@ def query_dataset(
     return dataset[index_map]  # type: ignore [this is fine]
 
 
+class QuerySlice:
+    def __init__(self, offset, size, dataset, query) -> None:
+        self.offset = offset
+        self.size = size
+        self.dataset = dataset
+        self.query = query
+
+    def slice_dataset(self):
+        return self.dataset[
+            self.offset : max(self.offset + self.size, len(self.dataset))
```

Comment:
Can offset be negative?  What happens if negative offset value is passed?

Reply:
No, offset always positive